src/ngraph/pass/auto_broadcast.hpp

```diff
@@ -0,0 +1,31 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include "ngraph/pass/pass.hpp"
+
+namespace ngraph
+{
+    namespace pass
+    {
+        class AutoBroadcast : public NodePass
```

Comment:
Maybe `ImplicitBroadcastElimination` is a better name?

Reply:
Also (no need to do for now, but) maybe we could have an option to specify which kinds of autobroadcast we do and don't want to eliminate. I think that some targets will support numpy broadcasting without elimination but might need Paddle-style to be converted to explicit.